backend/editor/api.py

```diff
@@ -35,4 +57,214 @@ async def pong(response: Response):
     Check server health
     """
     pong = datetime.now()
-    return {"ping": "pong @ %s" % pong}
\ No newline at end of file
+    return {"ping": "pong @ %s" % pong}
+
+@app.get("/nodesfull")
+async def findAllNodes(response: Response):
+    """
+    Get all nodes within taxonomy
+    """
+    query = """
+        MATCH (n) RETURN n
+    """
+    result = session.run(query)
+    allNodes = [record for record in result]
+    return allNodes
+
+@app.get("/entry/{entry}")
+async def findOneEntry(response: Response, entry: str):
+    """
+    Get entry corresponding to id within taxonomy
+    """
+    query = """
+        MATCH (n:ENTRY) WHERE n.id = $id 
+        RETURN n
+    """
+    result = session.run(query, {"id": entry})
+    oneEntry = [record for record in result]
+    return oneEntry

```

Comment:
you just want one entry. So it's better not to return the list.

I would write:
```suggestion
    oneEntry = next(iter(result))
    return oneEntry
```

but you also should handle the case where there is no such entry and return a 404.
```

Reply:
> you also should handle the case where there is no such entry and return a 404

Forgot to write that with proposed code it can be handled with a try / except.

But as you may also check you have only one item, you may also convert to list and the check list `len` and act accordingly.

Reply:
Would this check be required for the __header__ and __footer__ nodes too? Since in a taxonomy, it is understood that they are always present.

Reply:
Maybe not.